shared/js/background/chrome-events.es6.js

```diff
@@ -38,6 +45,118 @@ chrome.runtime.onInstalled.addListener(function (details) {
     }
 })
 
+/**
+ * ONBOARDING
+ * Logic to allow the SERP to display onboarding UI
+ */
+
+/**
+  * !! This function _can_ be sync (`cb` is called in the same tick) if the settings are alrady loaded
+  */
+function getOnboardingParams (cb) {
+    function onReady () {
+        cb(null, {
+            showWelcomeBanner: settings.getSetting('showWelcomeBanner'),
+            showCounterMessaging: settings.getSetting('showCounterMessaging')
+        })
+    }
+
+    if (settings.isReady()) {
```

Comment:
Cool thanks for highlighting this: https://app.asana.com/0/892838074342800/1200071625925781

This code could be made more JavaScript like with:
```
async function getOnboardingParams () {
    if (!settings.isReady()) {
        await settings.ready()
    }
    return {
        showWelcomeBanner: settings.getSetting('showWelcomeBanner'),
        showCounterMessaging: settings.getSetting('showCounterMessaging')
    }
}
```

With usage like:
```
const { showWelcomeBanner, showCounterMessaging } = await getOnboardingParams();
```

Reply:
Based on: https://app.asana.com/0/inbox/1199237043596759/1200071625925781/1200075771973335

This can be further shortened to:
```
async function getOnboardingParams () {
    await settings.ready()
    return {
        showWelcomeBanner: settings.getSetting('showWelcomeBanner'),
        showCounterMessaging: settings.getSetting('showCounterMessaging')
    }
}
```

Reply:
I have simplified this logic.